[Bug] rgb_matrix_update_pwm_buffers() is not explicitly declared

### Describe the Bug

When implementing the following example documented in https://docs.qmk.fm/custom_quantum_functions#example-shutdown-user-implementation:

```c
bool shutdown_user(bool jump_to_bootloader) {
    if (jump_to_bootloader) {
        // red for bootloader
        rgb_matrix_set_color_all(RGB_RED);
    } else {
        // off for soft reset
        rgb_matrix_set_color_all(RGB_OFF);
    }
    // force flushing -- otherwise will never happen
    rgb_matrix_update_pwm_buffers();
    // false to not process kb level
    return false;
}
```

Compiling a keymap with this code runs into:

```
./keyboards/massdrop/alt/keymaps/ericvw/keymap.c: In function 'shutdown_user':
./keyboards/massdrop/alt/keymaps/ericvw/keymap.c:73:5: error: implicit declaration of function 'rgb_matrix_update_pwm_buffers'; did you mean 'rgb_matrix_step_reverse'? [-Werror=implicit-function-declaration]
     rgb_matrix_update_pwm_buffers();
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     rgb_matrix_step_reverse
cc1: all warnings being treated as errors
```

### Keyboard Used

massdrop/alt

### Operating System

macOS and Windows

### qmk doctor Output

```text
> qmk doctor
Ψ QMK Doctor is checking your environment.
Ψ CLI version: 1.1.6
Ψ QMK home: /Users/eric/projects/qmk_firmware.git
Ψ Detected macOS 15.0 (Apple Silicon).
Ψ Userspace enabled: False
Ψ Git branch: massdrop/alt/ericvw
Ψ Repo version: 0.26.9
⚠ The official repository does not seem to be configured as git remote "upstream".
Ψ CLI installed in virtualenv.
Ψ All dependencies are installed.
Ψ Found arm-none-eabi-gcc version 8.5.0
Ψ Found avr-gcc version 8.5.0
Ψ Found avrdude version 8.0
Ψ Found dfu-programmer version 1.1.0
Ψ Found dfu-util version 0.11
Ψ Submodules are up to date.
Ψ Submodule status:
Ψ - lib/chibios: 2024-02-17 19:20:06 +0000 --  (be44b3305)
Ψ - lib/chibios-contrib: 2024-04-03 20:39:24 +0800 --  (77cb0a4f)
Ψ - lib/googletest: 2021-06-11 06:37:43 -0700 --  (e2239ee6)
Ψ - lib/lufa: 2022-08-26 12:09:55 +1000 --  (549b97320)
Ψ - lib/vusb: 2022-06-13 09:18:17 +1000 --  (819dbc1)
Ψ - lib/printf: 2022-06-29 23:59:58 +0300 --  (c2e3b4e)
Ψ - lib/pico-sdk: 2023-02-12 20:19:37 +0100 --  (a3398d8)
Ψ - lib/lvgl: 2022-04-11 04:44:53 -0600 --  (e19410f8)
Ψ QMK is ready to go, but minor problems were found
```

### Additional Context

I have a commit that changes `quantum/rgb_matrix.h` to declare `rgb_matrix_update_pwm_buffers` for a pull request if this is the desired change to address.

Alternatively, I am calling `rgb_matrix_driver.flush()`, which I'm not sure is desired to be called directly in keymaps.

--- a/keyboards/bastardkb/charybdis/charybdis.c
+++ b/keyboards/bastardkb/charybdis/charybdis.c
@@ -393,7 +393,6 @@ bool shutdown_kb(bool jump_to_bootloader) {
     rgblight_setrgb(RGB_RED);
 #endif // RGBLIGHT_ENABLE
 #ifdef RGB_MATRIX_ENABLE
-    void rgb_matrix_update_pwm_buffers(void);
     rgb_matrix_set_color_all(RGB_RED);
     rgb_matrix_update_pwm_buffers();
 #endif // RGB_MATRIX_ENABLE

--- a/keyboards/bastardkb/dilemma/dilemma.c
+++ b/keyboards/bastardkb/dilemma/dilemma.c
@@ -354,7 +354,6 @@ bool shutdown_kb(bool jump_to_bootloader) {
     rgblight_setrgb(RGB_RED);
 #endif // RGBLIGHT_ENABLE
 #ifdef RGB_MATRIX_ENABLE
-    void rgb_matrix_update_pwm_buffers(void);
     rgb_matrix_set_color_all(RGB_RED);
     rgb_matrix_update_pwm_buffers();
 #endif // RGB_MATRIX_ENABLE

--- a/keyboards/miiiw/blackio83/rev_0100/rev_0100.c
+++ b/keyboards/miiiw/blackio83/rev_0100/rev_0100.c
@@ -165,7 +165,6 @@ uint32_t loop_10Hz(uint32_t trigger_time, void *cb_arg) {
             wait_ms(50);
             eeconfig_init();
     #ifdef RGB_MATRIX_ENABLE
-            extern void rgb_matrix_update_pwm_buffers(void);
             for(int i = 0; i < 5; i++) {
                 rgb_matrix_set_color_all(RGB_WHITE);
                 rgb_matrix_update_pwm_buffers();

--- a/quantum/rgb_matrix/rgb_matrix.h
+++ b/quantum/rgb_matrix/rgb_matrix.h
@@ -210,6 +210,7 @@ void        rgb_matrix_decrease_speed_noeeprom(void);
 led_flags_t rgb_matrix_get_flags(void);
 void        rgb_matrix_set_flags(led_flags_t flags);
 void        rgb_matrix_set_flags_noeeprom(led_flags_t flags);
+void        rgb_matrix_update_pwm_buffers(void);
 
 #ifndef RGBLIGHT_ENABLE
 #    define eeconfig_update_rgblight_current eeconfig_update_rgb_matrix

